### PR TITLE
docs: add contributing guide (resolves #182)

### LINKED ISSUE #182: Add contributor guide

### PR DESCRIPTION
**Pull request by greatislander**
* [ ] This pull request has been linted by running `npm run lint` without errors
* [ ] This pull request has been tested by running `npm run dev` and reviewing affected routes
* [ ] This pull request has been built by running `npm run generate` without errors
* [ ] This isn't a duplicate of an existing pull request

## Description

Expands README to include contributing instructions.

## Related issues

- Resolves #182.

### CODE REVIEW

**Comment by cindyli** [comment]
@jobara's input on this pull request would be valuable. He's experienced dealing with with licensing, readme languages etc.

File: README.md [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
+
+Give your branch a descriptive name:
+
+- For a new feature, call it `feat/description-of-feature`
+- For a bug fix, call it `fix/description-of-bug`
+
+When committing your changes, use [Conventional Commits](https://conventionalcommits.org/).
+
+When your work is complete, open a pull request against the `dev` branch.
+
```
Comment (jobara) [inline README.md:44]: This should all go in to a CONTRIBUTING.md file. See: https://help.github.com/en/github/building-a-strong-community/setting-guidelines-for-repository-contributors

Other things that should be included here:
- merge method (I think we've been using squash). If we don't keep doing this, we should require that issue numbers be referenced in all commit logs.
- what's the workflow from moving from `dev` to `master`
- Other steps that should be performed before submitting a PR (e.g. linting, testing, etc)
- We should mention or point to any of the linting and other coding style requirements. 

We've been collecting a set of [project governance options](https://wiki.fluidproject.org/display/fluid/Project+Governance+Options). There may be some more things from there that could be added. Also feel free to add to that wiki doc as needed.

Comment (jobara) [reply README.md:44]: You can also look at the [CONTRIBUTING](https://github.com/fluid-project/infusion/blob/master/CONTRIBUTING.md) file from Infusion. You don't have to follow the same format; just providing as an example.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Give your branch a descriptive name:
+
+- For a new feature, call it `feat/description-of-feature`
+- For a bug fix, call it `fix/description-of-bug`
+
+When committing your changes, use [Conventional Commits](https://conventionalcommits.org/). Your commits will be automatically
+[linted](https://github.com/inclusive-design/wecount.inclusivedesign.ca#how-to-lint);
+you must address any linting errors before submitting a pull request.
+
+When your work is complete, open a pull request against the `dev` branch. Please make sure to fill out the pull request template.
```
Comment (jobara) [inline .github/CONTRIBUTING.md:27]: Should probably also make `dev` a link to the branch.

File: package.json [hunk 3]
```diff
@@ -21,7 +22,10 @@
     "lint-fix:css": "stylelint \"**/**.{scss,vue}\" --fix",
     "lint-fix:md": "markdownlint \"**/***.md\" --ignore-path .gitignore --ignore LICENSE.md --fix",
     "lint": "npm-run-all lint:js lint:css lint:md",
-    "lint:fix": "npm-run-all lint-fix:*",
+		"lint:fix": "npm-run-all lint-fix:*",
+		"release:major": "npx release major",
+		"release:minor": "npx release minor",
+		"release:patch": "npx release patch",
```
Comment (jobara) [inline package.json:28]: I don't see `release` as a dependency. Is it included by some other means? Also if it is installed as a dev dependency, I don't think we'd need to use `npx`.

File: package.json [hunk 4]
```diff
@@ -4,8 +4,9 @@
   "description": "The source files for the We Count website, built with Nuxt.",
   "license": "CC-BY-4.0",
   "contributors": [
-    "Ted Thompson <tthompson@ocadu.ca> (https://github.com/TeddTech)",
+		"Cindy Li <cli@ocadu.ca> (https://github.com/cindyli)",
     "Justin Obara <jobara@ocadu.ca> (https://github.com/jobara)",
+    "Ted Thompson <tthompson@ocadu.ca> (https://github.com/TeddTech)",
     "Ned Zimmerman <nzimmerman@ocadu.ca> (https://github.com/greatislander)"
   ],
   "private": true,
```
Comment (jobara) [inline package.json:12]: We've started using an [AUTHORS.md](https://github.com/fluid-project/infusion/blob/master/AUTHORS.md) file for other projects to Indicate those that have copyright. Probably at this point it doesn't matter, but something to think about for later. Also those of us at the IDRC all assign copyright to OCAD University.

File: README.md [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
 
-[![Netlify Status](https://api.netlify.com/api/v1/badges/d63b3d00-fd5f-47d7-8e43-d09bf4e8eb4f/deploy-status)](https://app.netlify.com/sites/wecount/deploys)
+[![License](https://img.shields.io/github/license/inclusive-design/wecount.inclusivedesign.ca?style=flat-square)](https://github.com/inclusive-design/wecount.inclusivedesign.ca/blob/dev/LICENSE.md)
+[![GitHub Release](https://img.shields.io/github/v/release/inclusive-design/wecount.inclusivedesign.ca?sort=semver&style=flat-square)](https://github.com/inclusive-design/wecount.inclusivedesign.ca/releases/latest)
+[![GitHub Workflow Status](https://img.shields.io/github/workflow/status/inclusive-design/wecount.inclusivedesign.ca/Test%20and%20build?label=github&style=flat-square)](https://github.com/inclusive-design/wecount.inclusivedesign.ca/actions)
+[![Netlify Status](https://img.shields.io/netlify/d63b3d00-fd5f-47d7-8e43-d09bf4e8eb4f?style=flat-square)](https://app.netlify.com/sites/wecount/deploys)
 
 The source files for the We Count website, built with [Nuxt](https://nuxtjs.org/).
 
 ## Getting Started
```
Comment (jobara) [inline README.md:10]: It would probably be good to have a link from somewhere in here to the CONTRIBUTING.md file.

File: package.json (hunk 4 above)
Comment (greatislander) [reply package.json:13]: Added an author field to package.json for now.

**Comment by greatislander** [comment]
> I'm struggling a bit with the dev content in the README vs the CONTRIBUTING file, but can't quite put my finger on it. I get the feeling like I need to go back and forth between the two documents. Maybe that's okay, but not sure. If you can think of any further ways to improve this, please add them otherwise I guess we can leave it for later.

My differentiation strategy is: technical guide in README, process guide in CONTRIBUTING. We can always adjust/improve later though, as you say.

> Regarding the "release" tool. I believe it saves the release info straight to GitHub.

When you run release, it:

- bumps the version based on the semantic versioning strategy (patch, minor, or major)
- provides an interactive prompt which walks you through each commit since the last tag to build a changelog
- commits, tags, and pushes the new version to GitHub
- opens a draft GitHub release with the generated changelog ([example](https://github.com/inclusive-design/wecount.inclusivedesign.ca/releases/tag/1.0.1))

>     1. can we automate this with GitHub Actions. I suppose we might have to do some determining of the type of release if there isn't an API for that.

Yes, this is definitely something we can explore but feels low priority at the moment.

>     2. if we don't use GitHub actions, do we want to include this info in the readme or a change log that's stored in the repo and included in the tag

I think the GitHub release notes are sufficient but we can look into storing them in a changelog as well, but I'm happy with the approach that is in place right now.

**Comment by jobara** [comment]
> I think the GitHub release notes are sufficient but we can look into storing them in a changelog as well, but I'm happy with the approach that is in place right now.

There's an [issue](https://github.com/zeit/release/issues/121) related to this filed against release. Adding here for reference.

**Comment by jobara** [comment]
I couldn't figure out a way to test the release scripts, but they looked correct. On the bright side, if they don't work, I guess it won't publish the release. So the risk seems minimal.

File: package.json [hunk 6]
```diff
@@ -2,10 +2,12 @@
   "name": "@wecount/wecount.inclusivedesign.ca",
   "version": "1.0.1",
   "description": "The source files for the We Count website, built with Nuxt.",
-  "license": "CC-BY-4.0",
+	"license": "CC-BY-4.0",
+	"author": "OCAD University <idrc@ocadu.ca>",
```
Comment (jobara) [inline package.json:6]: @greatislander I think this is the last thing, there are mixed tabs and spaces in this file.

**Pull request merged**